src/Microsoft.ComponentDetection.Common/DirectoryUtilityService.cs

```diff
@@ -0,0 +1,51 @@
+namespace Microsoft.ComponentDetection.Common;
+
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.ComponentDetection.Contracts;
+
+/// <inheritdoc />
+public class DirectoryUtilityService : IDirectoryUtilityService
+{
+    /// <inheritdoc />
+    public void Delete(string path, bool recursive) => Directory.Delete(path, recursive);
+
+    /// <inheritdoc />
+    public IEnumerable<string> EnumerateDirectories(string path) =>
+        Directory.EnumerateDirectories(path);
+
+    /// <inheritdoc />
+    public IEnumerable<string> EnumerateDirectories(string path, string searchPattern, SearchOption searchOption) =>
+        Directory.EnumerateDirectories(path, searchPattern, searchOption);
+
+    /// <inheritdoc />
+    public IEnumerable<string> EnumerateFiles(string path, string searchPattern, SearchOption searchOption) =>
+        Directory.EnumerateFiles(path, searchPattern, searchOption);
+
+    /// <inheritdoc />
+    public bool Exists(string directoryPath) => Directory.Exists(directoryPath);
+
+    /// <inheritdoc />
+    public (HashSet<string> Files, HashSet<string> Directories) GetFilesAndDirectories(string root, IList<string> patterns, int depth)
+    {
+        var fileList = new List<string>();
```

Comment:
I included a depth here to avoid the infinite loop that can be introduced when AllDirectories is used: https://learn.microsoft.com/en-us/dotnet/api/system.io.searchoption?view=net-8.0#remarks 

It also reduces the amount of computation that happens for each of the detectors, and so far I haven't seen a directory created more than 2 levels below the `setup.py` files